Add support of static IPv6.

cathay4t opened the pull request:
Example:

```json
{
    "interfaces": [
        {
            "ipv4": {
                "address": [
                    {
                        "ip": "192.168.32.11",
                        "prefix-length": 24
                    }
                ],
                "enabled": true
            },
            "ipv6": {
                "address": [
                    {
                        "ip": "fe80::8097:51a0:9ff4:c7a1",
                        "prefix-length": 64
                    }
                ],
                "enabled": true
            },
            "mtu": 1500,
            "name": "wlp3s0",
            "state": "up",
            "type": "unknown"
        }
    ]
}
```

cathay4t commented:
Working on the test cases. Please hold the code review.

EdDev commented on libnmstate/nm/ip.py:
```diff
@@ -0,0 +1,76 @@
+from . import nmclient
+
+
+def get_ip_info(active_connection, is_ipv6=False):
```
Boolean flags imply that this function is doing more than one thing.
It is preferred to use separated functions.
The common code can be called from both such functions.

cathay4t commented:
V2 patch read for review.

cathay4t commented:
Thanks for the inputs. I will move those codes back to ipv4 and ipv6 modules.

cathay4t commented:
V3 patch uploaded.
Changes since V2:
 * Dropped the `uitls/_iplib.py` and moved codes back to `ipv4.py` and `ipv6.py`.

EdDev commented on libnmstate/nm/ipv6.py:
```diff
@@ -15,10 +15,46 @@
+        return info
+
+    ipconfig = active_connection.get_ip6_config()
+
```
No need for newline, they are related.

EdDev commented on libnmstate/nm/ipv6.py:
```diff
@@ -15,10 +15,46 @@
 #
 
-from . import nmclient
+import socket
```
Just a convention I prefer: Please separate std, 3rd party and internal packages using a newline.

EdDev commented on tests/lib/nm/ipv4_test.py:
```diff
@@ -24,7 +25,7 @@
 @pytest.fixture
 def NM_mock():
-    with mock.patch.object(nm.ipv4.nmclient, 'NM') as m:
+    with mock.patch.object(nm.nmclient, 'NM') as m:
```
We need to mock at the usage not at the original definition.
These change also also not related to this PR/commit, you can propose it in another context if you think it is needed.

cathay4t replied:
Sure. Just old left up. Will clean them up.

EdDev commented on tests/lib/nm/ipv6_test.py:
```diff
@@ -15,15 +15,106 @@
-    ipv6_setting = nm.ipv6.create_setting()
+@pytest.fixture
+def NM_mock():
+    with mock.patch.object(nm.nmclient, 'NM') as m:
```
Same comment as the one in the IPv4 test module.

EdDev commented on tests/lib/nm/ipv4_test.py:
```diff
@@ -16,6 +16,7 @@
 #
 
 import pytest
+import socket
```
Why not leave it up to the ipv4 module to provide it? What is the advantage of exposing the test module to this dependency?
Same for the IPv6 test module.

cathay4t commented:
V4 patch uploaded. 
Changes since V3:  
 * Remove some old left up test changes.
 * Fix the code style choice on separating standard module imports and non-standard ones.
